src/language-js/utils.js

```diff
@@ -895,6 +1006,7 @@ function isSimpleCallArgument(node, depth) {
   const plusTwo = (node) => isSimpleCallArgument(node, depth + 2);
 
   const regexpPattern =
+    // @ts-ignore
```

Comment:
This `@ts-ignore` does not seem to be needed if we do this to `src/language-js/types/estree.d.ts`:

```diff
-export type Node = ESTree.Node | Babel.Node | TSTree.Node | NGTree.NGNode;
+export type Node = Babel.Node | TSTree.Node | NGTree.NGNode;
```

Reply:
`ESTree.Node` stays see https://github.com/prettier/prettier/pull/8759#discussion_r467690623 , you  might  not  noticed we  didn't add Flow Node, because `estree/flow.d.ts` is  broken as I tried.

Reply:
> see [#8759 (comment)](https://github.com/prettier/prettier/pull/8759#discussion_r467690623)

says this, for the sake of easier reference for myself:

> `estree` Literal can have `regex` property.
> 
> https://github.com/estree/estree/blob/master/es5.md#regexpliteral

Here are the type errors I get if we keep `ESTree.Node`:

```console
src/language-js/utils.js:1013:38 - error TS2339: Property 'regex' does not exist on type 'SimpleLiteral | RegExpLiteral | BigIntLiteral | BooleanLiteral | NumberLiteral | NullLiteral | RegExpLiteral | StringLiteral'.
  Property 'regex' does not exist on type 'SimpleLiteral'.

1013     (node.type === "Literal" && node.regex && node.regex.pattern) ||
                                          ~~~~~

src/language-js/utils.js:1013:52 - error TS2339: Property 'regex' does not exist on type 'SimpleLiteral | RegExpLiteral | BigIntLiteral | BooleanLiteral | NumberLiteral | NullLiteral | RegExpLiteral | StringLiteral'.
  Property 'regex' does not exist on type 'SimpleLiteral'.

1013     (node.type === "Literal" && node.regex && node.regex.pattern) ||
                                                        ~~~~~
```

This type issue sure looks ugly to me.

TBH I still don't really understand the need for `ESTree.Node`, considering the Node types would be mostly if not completely covered by Babel and TSTree. But I can put it back, maybe with a TODO comment, if needed.

One more thing is that I am not sure if I follow what language-js does with Angular. I can see that language-js deals with Angular embedded in a Component decorator factory, am I missing anything else major here?

Reply:
> Here are the type errors.

I mean something wrong in other place.

Reply:
> I still don't really understand the need for ESTree.Node

Because `ESTree.Node` is `Node`, we'll use it in other places.

Reply:
> One more thing is that I am not sure if I follow what language-js does with Angular. I can see that language-js deals with Angular embedded in a Component decorator factory, am I missing anything else major here?

I don't understand this question

Reply:
I just reverted my change and put `ESTree.Node` back, and put the `@ts-ignore` comment back as well. I am very hopeful that we can resolve the type issue someday.

And my question was simply to understand what language-js does with Angular. I can research it another day.

Please let me know if there is anything else needed.

Reply:
`angular-estree-parser` is also a js parser https://github.com/prettier/prettier/blob/6833310f87c597f0a204b06f2e84cb1b3b33ff54/src/language-js/parser-angular.js#L7